pyshelf/metadata/container.py

```diff
@@ -0,0 +1,59 @@
+from pyshelf.metadata.mapper import Mapper
+from pyshelf.metadata.wrapper import Wrapper
+from pyshelf.metadata.manager import Manager
+from pyshelf.metadata.yaml_codec import YamlCodec
+from pyshelf.metadata.cloud_portal import CloudPortal
+from pyshelf.metadata.initializer import Initializer
+
+
+class Container(object):
+    def __init__(self, bucket_name, cloud_factory, resource_identity):
+        self.bucket_name = bucket_name
+        self.cloud_factory = cloud_factory
+        self.resource_identity = resource_identity
+        self._mapper = None
+        self._manager = None
+        self._yaml_codec = None
+        self._cloud_portal = None
+        self._initializer = None
+
+    def create_cloud_storage(self):
+        return self.cloud_factory.create_storage(self.bucket_name)
+
+    def create_wrapper(self, metadata):
+        return Wrapper(metadata)
+
+    @property
+    def mapper(self):
+        if not self._mapper:
+            self._mapper = Mapper()
+
+        return self._mapper
+
+    @property
+    def manager(self):
```

Comment:
I think it would be nice to add what it returns (full class path).  As far as a description, if we do add descriptions, I think they should go in the actual class file